File: src/main/java/com/jore/epoc/bo/EpocCalendar.java
package com.jore.epoc.bo;

import java.time.DayOfWeek;
import java.time.LocalDate;

public class EpocCalendar {
    private static EpocCalendar INSTANCE = new EpocCalendar();

    public static EpocCalendar getInstance() {
        return INSTANCE;
    }

    public boolean isWorkingDay(LocalDate date) {
        return !date.getDayOfWeek().equals(DayOfWeek.SATURDAY) && !date.getDayOfWeek().equals(DayOfWeek.SUNDAY);
    }
}

File: EpocServer/Sources/App/Foundation/EpocCalendar.swift
import Foundation

/// Utility for working day calculations. Equivalent to `com.jore.epoc.bo.EpocCalendar`.
public struct EpocCalendar: Sendable {
    public static let shared = EpocCalendar()

    private let calendar: Calendar = {
        var cal = Calendar(identifier: .gregorian)
        cal.timeZone = TimeZone(identifier: "UTC")!
        return cal
    }()

    /// Returns true if the given date is a working day (not Saturday or Sunday).
    public func isWorkingDay(_ date: Date) -> Bool {
        let weekday = calendar.component(.weekday, from: date)
        // In Calendar: 1 = Sunday, 7 = Saturday
        return weekday != 1 && weekday != 7
    }

    /// Returns all working days in the given YearMonth.
    public func workingDays(in yearMonth: YearMonth) -> [Date] {
        yearMonth.daysInMonth().filter { isWorkingDay($0) }
    }

    /// Returns the number of working days in the given YearMonth.
    public func workingDayCount(in yearMonth: YearMonth) -> Int {
        workingDays(in: yearMonth).count
    }
}
